test/e2e/Command/AssertBackwardsCompatibleTest.php

```diff
@@ -0,0 +1,301 @@
+                '--from=' . $this->versions[0],
+                '--to=' . $this->versions[1],
+            ],
+            $this->sourcesRepository
+        );
+
+        self::assertSame(3, $check->run());
+        self::assertStringEndsWith(
+            <<<'EXPECTED'
+[BC] CHANGED: The parameter $a of TestArtifact\TheClass#method() changed from TestArtifact\A to a non-contravariant TestArtifact\B
+1 backwards-incompatible changes detected
+
+EXPECTED
+            ,
+            $check->getErrorOutput() // @TODO this looks like a symfony bug - we shouldn't check STDERR, but STDOUT
+        );
+    }
+
+    public function testWillNotRunWithoutTagsNorSpecifiedVersions() : void
+    {
+        $check = new Process(
+            __DIR__ . '/../../../bin/roave-backward-compatibility-check',
+            $this->sourcesRepository
+        );
+
+        self::assertSame(212, $check->run());
+        self::assertContains(
+            'Could not detect any released versions for the given repository',
+            $check->getErrorOutput()
+        );
+    }
+
+    public function testWillRunSuccessfullyOnNoBcBreaks() : void
+    {
+        $check = new Process(
+            [
+                __DIR__ . '/../../../bin/roave-backward-compatibility-check',
+                '--from=' . $this->versions[0],
+                '--to=' . $this->versions[3],
+            ],
+            $this->sourcesRepository
+        );
+
+        self::assertSame(0, $check->run());
+        self::assertContains(
+            'No backwards-incompatible changes detected',
+            $check->getErrorOutput()
+        );
+    }
+
+    public function testWillPickTaggedVersionOnNoGivenFrom() : void
+    {
+        (new Process(
+            [
+                'git',
+                'checkout',
+                $this->versions[1],
+            ],
+            $this->sourcesRepository
+        ))->mustRun();
+        (new Process(
+            [
+                'git',
+                'tag',
+                '1.2.3',
+                '-m',
+                'First tag',
+            ],
+            $this->sourcesRepository
+        ))->mustRun();
+
+        $check = new Process(
+            [
+                __DIR__ . '/../../../bin/roave-backward-compatibility-check',
+                '--to=' . $this->versions[2],
+            ],
+            $this->sourcesRepository
+        );
+
+        self::assertSame(3, $check->run());
+
+        $errorOutput = $check->getErrorOutput();
+
+        self::assertContains('Detected last minor version: 1.2.3', $errorOutput);
+        self::assertStringEndsWith(
+            <<<'EXPECTED'
+[BC] CHANGED: The parameter $a of TestArtifact\TheClass#method() changed from TestArtifact\B to a non-contravariant TestArtifact\C
+1 backwards-incompatible changes detected
+
+EXPECTED
+            ,
+            $errorOutput // @TODO this looks like a symfony bug - we shouldn't check STDERR, but STDOUT
```

Comment:
This is a problem. For some reason, everything is going to STDERR in my tests (tried manually via `bash`, so the bug is with `symfony/console` or with how we use it)

Reply:
https://github.com/symfony/symfony/pull/13730 ?

Reply:
Looked it up as well - unsure if that's the actual bug. I'll probably need to dump the streams and see if both point to STDERR by accident, because the output sure only is on STDERR.

Reply:
Umm, we are manually splitting output between stderr/stdout ourselves.

If you specify `--format=markdown`, `stdout` is formatted markdown: `stderr` contains the human-readable output, so this is the expectation. Not a symfony bug.

See these for `stderr` being used:

https://github.com/Roave/BackwardCompatibilityCheck/blob/9e95a606a84f398212b17287aa7801d39c72e460/src/Command/AssertBackwardsCompatible.php#L146

https://github.com/Roave/BackwardCompatibilityCheck/blob/9e95a606a84f398212b17287aa7801d39c72e460/src/Command/AssertBackwardsCompatible.php#L151-L153

https://github.com/Roave/BackwardCompatibilityCheck/blob/9e95a606a84f398212b17287aa7801d39c72e460/src/Command/AssertBackwardsCompatible.php#L185

And then, markdown formatting if required:

https://github.com/Roave/BackwardCompatibilityCheck/blob/9e95a606a84f398212b17287aa7801d39c72e460/src/Command/AssertBackwardsCompatible.php#L190-L192

This was done specifically so you can pipe output to a file - see PR #35 , specifically https://github.com/Roave/BackwardCompatibilityCheck/pull/35#discussion_r181588762

Reply:
Yes, but in the test (also manual ones), redirecting output doesn't seem to have any effect

Reply:
Moved to https://github.com/Roave/BackwardCompatibilityCheck/issues/79